### ISSUE TITLE
[Landing Page] Redirect workspace members invited on NewDot

### ISSUE THREAD

**Issue by trjExpensify** [Expensify/App#45015]
Coming from [here](https://expensify.slack.com/archives/C036QM0SLJK/p1719577689520919?thread_ts=1719577680.548439&cid=C036QM0SLJK). 

### Problem 

We're working towards redirecting 100% of sign-ups to NewDot with the Summer release, yet the people these sign-ups invite to a workspaces can end up with a different experience to the one they're using themselves. 

### Why is this important? 

It creates a confusing dynamic for the inviting admin and their workspace members, as they'll be blissfully unaware of this as they're new to Expensify. For example, if the admin gives the member instructions on how to do something that they can't see, like a reference to a place in the UI that doesn't exist in the platform they're on etc.

### Solution 

Default to NewDot any new account invited to a workspace on NewDot. This means setting `classicRedirect.dismissed` to `false` in the `tryNewDot` NVP for these invited workspace members. In turn, if they head to expensify.com, they'll be redirected to new.expensify.com. If they download the (then released) HybridApp, we'll load the NewDot native experience, not the OldApp one. 

For the avoidance of doubt, any new account invited to a workspace via OldDot should **_not_** be defaulted to NewDot at this stage. 

____
CC: @Julesssss @JmillsExpensify @danielrvidal @puneetlath @AndrewGable from the thread. I'm going to track this with the landing page project in #wave-collect, so we have it done at the latest by then when we'll start turning on the tap for all new sign-ups on expensify.com being routed to new.expensify.com. But as discussed in thread, nothing is stopping us doing this right now.

**Comment by Julesssss** [comment]
Do we care about this edge case?
- The invited user is an admin of an existing OldDot policy?

**Comment by trjExpensify** [comment]
Hm, I don't think so. If you're being invited via NewDot let's default you to NewDot.

**Comment by JmillsExpensify** [comment]
Agree with Tom

**Comment by Julesssss** [comment]
Today I Implemented the invitation logic, refactored the redirection helper functions, and added automatic test cases. Remaining steps for tomorrow morning: 

- Fix a single failing test case
- Write manual test cases and submit the PR for review
- Update the StackOverflow to represent the new logic

[WIP here](https://github.com/Expensify/Web-Expensify/pull/42760)

**Comment by Julesssss** [comment]
[PR submitted for review](https://github.com/Expensify/Web-Expensify/pull/42760)

**Comment by Julesssss** [comment]
Answered some questions on my PR, I think it should be good to merge today 🤞

**Comment by Julesssss** [comment]
Merged, awaiting deployment to prod. Once merged I will:
- Submit [these](https://github.com/Expensify/Expensify/issues/405342) new QA Testrails cases
- Update the Stackoverflow for redirection logic

**Comment by trjExpensify** [comment]
Excellent!

**Comment by Julesssss** [comment]
@trjExpensify one thing worth mentioning is that users navigating to `expensify.com/inbox` will not be redirected to NewDot. I believe this was done on purpose, but we might want to force authentication to drop the `/inbox` route to ensure users are always dropped into NewDot...

**Comment by trjExpensify** [comment]
@Julesssss I'm trying to think when would they navigate to expensify.com/inbox in this workspace invite flow? 🤔

**Comment by Julesssss** [comment]
I could be wrong about this. But I feel like there is some logic in the backend that routes users to /inbox the first time their account logs in, or something like that.

During testing today it was fine 80% of the time, but once or twice I was taken to Inbox and I'm 90% sure I had navigated to `staging.expensify.com` to sign in.

**Comment by trjExpensify** [comment]
Interesting. So for a NewDot workspace invite we've got clicking the URL of the report in the footer of one of these two emails as the likely primary flow:

<img width="1202" alt="image" src="https://github.com/user-attachments/assets/2de47dea-929a-4570-9176-94c37a0ff4a4">

.. both of which are new.expensify.com or email.expensify.com links that validate and go to NewDot. 

Then in the case where you're invited via NewDot, but just head to expensify.com to try and sign-in for the first time. You're prompted for a magic code, enter that, and it's routing to NewDot now accordingly (based on your work):

https://github.com/user-attachments/assets/0ada716d-090f-4ffe-ad02-9f4c82646e1a

So yeah, I'm not sure where you would have expensify.com/inbox as the URL to enter the product? 🤔

**Comment by Julesssss** [comment]
Alright, thanks for double-checking. That's good to hear and I'm happy to mark this as complete and forget about it in the meantime. 

But in case we ever come back to this, [this](https://github.com/Expensify/Web-Expensify/blob/bca153a62d2d2e06f2015ea5dcdd637bb204ec9d/homepage/js/signin.jsx#L1611) is the line of code that I believe is sometimes routing me to inbox. Perhaps it only occurs for users with expired authtokens, or those navigating to an invalid route 🤷 

```
// We default exitTo to '/inbox' so that it attempts to use the newly acquired authToken if no exitTo is supplied, instead of going to /undefined
```

**Comment by trjExpensify** [comment]
Oh interesting, okay. That code block is helpful. 

>// If we have the infinite session credentials stored in the browser, then show an authenticating interim page to the user
            $('#authenticating').removeClass('hidden');

So with infinite sessions, if you navigate to expensify.com (with an expired authToken), we show the authenticating interim screen (while we generate you a new one?) and then exitTo `expensify.com/inbox`. Is that right? 

In which case, you're saying we wouldn't redirect "eligible users" to new.expensify.com on a returning visit with an expired authToken? Unless we do a check for classicRedirect.dismissed===true/false before that exitTo redirect to /inbox is executed? 

So this would be a pitfall of the redirect logic for on-going use and returning users, basically? If so, I do think we should probably plug that gap. Otherwise (if I'm following this correctly) we have this going on:

1. Sign-up flows (i.e welcome email, unread summary email, navigating to expensify.com natively for the first time to sign-in), you get redirected. 
2. Returning use with a valid authToken, you get redirected.
3. Returning use with an expired authToken, you don't get redirected. 

Let me know if I'm not catching your drift there.

**Comment by Julesssss** [comment]
Yeah, I agree 100%. It could be very rare, but I think patching makes sense. Of course, this is assuming NewDot users being able to view the OldDot inbox is expected behaviour.

**Comment by trjExpensify** [comment]
>Of course, this is assuming NewDot users being able to view the OldDot inbox is expected behaviour.

Do you mean **not** expected here? But either way, I think more so the expectation can be stated as: 

When a returning user that has `classicRedirect.dismissed` set to `false` returns to expensify.com and re-authenticates, they're redirected to NewDot thereafter, not the OldDot inbox.

**Comment by Julesssss** [comment]
> Do you mean **not** expected here?

**A)** Users with the NVP re-authenticatating are not redirected to NewDot when navigating to OldDot
**B)** Users with the NVP navigating to `expensify.com/inbox` do not get redirected

I was talking about **B**.  I assume this is expected behaviour:
- Navigating to `expensify.com` > redirected
- Navigating to `expensify.com/inbox` > not redirected, as they need a specific inbox feature

______

Regardless. I'm on board with you.

**Comment by trjExpensify** [comment]
Yep, agreed that's expected.

**Comment by trjExpensify** [comment]
@Julesssss we can close this, right?

**Comment by Julesssss** [comment]
> Default to NewDot any new account invited to a workspace on NewDot. This means setting classicRedirect.dismissed to false in the tryNewDot NVP for these invited workspace members. In turn, if they head to expensify.com, they'll be redirected to new.expensify.com.

I implemented this, but it has since broken with some regression. So lets keep this issue open?

**Comment by trjExpensify** [comment]
Remind me of the regression? :/

**Comment by Julesssss** [comment]
Previously, the TryNewDot NVP would be set for new users who sign up using new.expensify.com or NewApp. 

At some point this has stopped:

> Default to NewDot any new account invited to a workspace on NewDot.

**Comment by Julesssss** [comment]
After retesting I think this is the current state: 
- NVP applied after invitation: Working ✅ 
- NVP applied after natural account creation: Not working ❌ 

**'Invitation' Tested here:**
<img width="275" alt="Screenshot 2024-08-12 at 10 01 51" src="https://github.com/user-attachments/assets/e54e72c5-ee54-458d-bb1c-c33df2a12b2b">
<img width="327" alt="Screenshot 2024-08-12 at 10 01 56" src="https://github.com/user-attachments/assets/0e872389-9263-4670-b77e-fe1650a0fb42">
<img width="372" alt="Screenshot 2024-08-12 at 10 02 21" src="https://github.com/user-attachments/assets/cc74bc13-6a1e-4d5d-bd42-d5debc22badb">

**Comment by Julesssss** [comment]
> we can close this, right?

Sorry, yeah I agree with this. We still need to do [this](https://github.com/Expensify/Expensify/issues/412892) I believe.
